index.js

```diff
@@ -230,17 +224,67 @@ module.exports.processV3 = (output, origin, locales = []) => {
   };
 };
 
+module.exports.processV4 = (output, origin, locales = []) => {
+  const data = JSON.parse(output.toString().trim());
+  const { automatic_captions, formats, subtitles } = data;
+  const cookies = data.http_headers && data.http_headers.Cookie || '';
+  const duration = data.duration || 0;
+  const subtitleFile = findBestSubtitleFile(subtitles, locales) || findBestSubtitleFile(automatic_captions, locales);
+  const subtitleUrl = subtitleFile ? `${origin}/ytdl/vtt?suburi=${encodeURIComponent(subtitleFile.subs.url)}` : '';
+  const title = data.title || '';
+  const processedVideoTracks = processVideoFormats(formats, !data.duration);
+  const thumbnail = data.thumbnail || '';
+
+  const audioTracks = processAudioFormats(formats, true);
+
+  const video_tracks = processedVideoTracks.map(formatInfo => {
+    if (formatInfo.fragments) {
+      const manifest = generateManifest({ ...formatInfo, duration });
+      return { type: 'manifest', manifest, height: formatInfo.height, combined: formatInfo.acodec !== 'none', format_id: formatInfo.format_id, protocol: formatInfo.protocol };
+    } else {
+      return { type: 'url', url: formatInfo.url, height: formatInfo.height, combined: formatInfo.acodec !== 'none', format_id: formatInfo.format_id, protocol: formatInfo.protocol };
+    }
+  });
+
+  // In V4 we just return all the eligible audio tracks and let the box pick
+  // This is so that gstreamer can pick a m3u8 track and Vivi Anywhere can pick a non-m3u8 track
+  let silent_video = false;
+  const formatttedTracks = audioTracks.map(audioTrack => {
+    const { acodec, fragments: audio_fragments, url: audio_url, format_id: audio_format, abr: audio_bitrate, protocol: audio_protocol, language } = audioTrack;
+    const audio_language = language || 'unknown';
+    if (isSilentVideo(audio_bitrate)) {
+      silent_video = true;
+    } else if (audio_fragments) {
+      const audioManifest = generateManifest(data, true);
+      return { type: 'manifest', acodec, manifest: audioManifest, format_id: audio_format, protocol: audio_protocol, language: audio_language };
+    } else {
+      return { type: 'url', acodec, url: audio_url, format_id: audio_format, protocol: audio_protocol, language: audio_language };
+    }
+  });
+
+  return {
+    cookies,
+    duration,
+    subtitle_url: subtitleUrl,
+    title,
+    thumbnail,
+    audio: formatttedTracks,
+    video: video_tracks,
+    silent_video
+  };
+};
+
 module.exports.processPlaylist = (output) => {
-  const outputJSON = JSON.parse(output)
-  if (outputJSON["entries"]) {
-    return outputJSON["entries"].map((video) => `https://www.youtube.com/watch?v=${video.id}`);
+  const outputJSON = JSON.parse(output);
+  if (outputJSON['entries']) {
+    return outputJSON['entries'].map((video) => `https://www.youtube.com/watch?v=${video.id}`);
```

Comment:
```suggestion
  if (outputJSON.entries) {
    return outputJSON.entries.map((video) => `https://www.youtube.com/watch?v=${video.id}`);
```
Would be good to lint against this too

Reply:
Done